### DISCUSSION TITLE
Is possible change youtube video alignment (left/center/right)?

### DISCUSSION THREAD

**Discussion by kaiunido** [ueberdosis/tiptap#4827] [General Questions] [answered]
Using textalign-extension with youtube-extension if you config youtube video to be inline and try to align to center, in the editor works fine but after save and reload the page or use in the Vue v-html (Chrome, don't know about other browsers) remove `<p>` tag wrapping video because `div` tag inside `p` is not a valid HTML (`div` is not a [phrasing content](https://html.spec.whatwg.org/multipage/dom.html#phrasing-content)).

If you remove inline config from youtube extension you can't align video using textalign-extension.

With `<p>` from request:
`<p style=\"text-align: center\"><div data-youtube-video=\"\"><iframe width=\"640\" height=\"480\" allowfullscreen=\"true\" autoplay=\"false\" disablekbcontrols=\"false\" enableiframeapi=\"false\" endtime=\"0\" ivloadpolicy=\"0\" loop=\"false\" modestbranding=\"false\" origin=\"\" playlist=\"\" src=\"https:\/\/www.youtube.com\/embed\/......\" start=\"0\"><\/iframe><\/div><\/p>`

After `v-html`:
![image](https://github.com/ueberdosis/tiptap/assets/5544018/d00baa7f-edca-46bb-b869-41b717e07a82)

There's someway to achieve this using textalign-extension to use the same alignment buttons?

Thank you.

**Comment by thefinnomenon** [comment]
You can't do it with `text-align` but you can add new buttons or edit the buttons to set `float` for the youtube video instead.

**Reply by kaiunido** [reply, marked as answer]
Thank you so much for your answer and sorry for my late reply.

I tried to achieve what you said with float, but I think my knowledge about the library is very limited to get the node and add a custom style on the fly. For a moment I finded a way to get the nodeDOM and apply the style but it don't reflect to nodes from Tiptap and after save the edition go away.

After that, I checked the extension code and try extend adding a display flex and adding a new attr to set the flex aligment and it works. I don't know if is the best or the easiest way but, if helps someone trying to do the same, here is the code...

Extending Youtube extension:
```ts
const CustomYoutube = Youtube.extend({
  addAttributes() {
    return {
      ...this.parent?.(),
      flexAlign: {
        default: 'flex-start',
      },
    };
  },

  // @ts-ignore
  renderHTML(attributes) {
    const newRender = this.parent?.(attributes) as unknown as unknown[];

    if (newRender && newRender?.[1]) {
      newRender[1] = {
        ...newRender[1],
        style: `display: flex; justify-content: ${attributes.node.attrs.flexAlign}`,
      };
    }

    return newRender;
  },
});
```
I used "unknown" types and "ts-ignore" because I don't know where to get the types.
And parent type "DOMOutputSpec" give me type errors when I tried to use array index (`newRender?.[1]`) as "same as any".

Add to Editor config (I'm using composable):
```ts
const editor = useEditor({
  // ...
  extensions: [
    // ...
    CustomYoutube.configure(),
  ],
});
```

And create a method to call on button click event:
```ts
const handleAlignmentClick = (alignment = 'left') => {
  if (!editor.value) return;

  const { from } = editor.value.state.selection;
  const node = editor.value.state.doc.nodeAt(from);

  if (node && node.type.name === 'youtube') {
    switch (alignment) {
      case 'center':
        editor.value.commands.updateAttributes('youtube', {
          flexAlign: 'center',
        });
        break;
      case 'right':
        editor.value.commands.updateAttributes('youtube', {
          flexAlign: 'flex-end',
        });
        break;
      default:
        editor.value.commands.updateAttributes('youtube', {
          flexAlign: 'flex-start',
        });
    }

    return;
  }

  editor.value.chain().focus().setTextAlign(alignment).run();
};
```
